dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/LogsChangeTraceSpecTest.java

```diff
@@ -47,26 +50,51 @@ public class LogsChangeTraceSpecTest {
     public static void setUpClass() throws Exception {
         ShrinkHelper.defaultApp(server, APP_NAME, new DeployOptions[] { DeployOptions.OVERWRITE }, "app2");
         FATSuite.copyAppsAppToDropins(server, APP_NAME);
+
+        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, false, null);
+        server.startServer();
     }
 
     @Before
-    public void startServer() throws Exception {
-        server.setCheckpoint(CheckpointPhase.AFTER_APP_START, false, null);
-        server.startServer(getTestMethodNameOnly(testName) + ".log");
+    public void setArchiveMarker() throws Exception {
+        server.setArchiveMarker(getTestMethodNameOnly(testName) + ".marker");
     }
 
     @After
     public void tearDown() throws Exception {
         server.stopServer();
-        server.deleteAllDropinConfigurations();
+        // There should be trace and messages after stopping
+        try {
+            checkForRestoreLogDir();
+        } finally {
+            server.deleteAllDropinConfigurations();
+            server.deleteDirectoryFromLibertyServerRoot("logs");
+            server.deleteDirectoryFromLibertyServerRoot(RESTORE_LOG_DIR);
+            server.setLogsRoot(server.getServerRoot() + File.separator + "logs" + File.separator);
+        }
     }
 
     @Test
-    public void testSetTraceSpec() throws Exception {
+    public void testSetTraceSpecBeforeRestore() throws Exception {
         server.addDropinOverrideConfiguration("dropinConfigChange/enableCheckpointTrace.xml");
         // Must set the logs root to the configured one; otherwise the check for successful restore will fail
-        server.setLogsRoot(server.getServerRoot() + File.separator + "restore_log_dir" + File.separator);
+        server.setLogsRoot(server.getServerRoot() + File.separator + RESTORE_LOG_DIR + File.separator);
         server.checkpointRestore();
     }
 
+    @Test
+    public void testSetTraceSpecAfterRestore() throws Exception {
+        server.checkpointRestore();
+        server.addDropinOverrideConfiguration("dropinConfigChange/enableCheckpointTrace.xml");
+        server.setLogsRoot(server.getServerRoot() + File.separator + RESTORE_LOG_DIR + File.separator);
+        // wait a bit to make sure file monitor runs
+        Thread.sleep(1000);
+    }
+
+    private void checkForRestoreLogDir() throws Exception {
```

Comment:
Did you want to assert any condition about the logs created during checkpoint?

Reply:
This test is about changing the log directory on restore.  There is no need to check the logs created on the checkpoint side because the verify steps for a successful checkpoint in the `LibertyServer` class already read from the logs on checkpoint from the expected, original, location.